modules/core/shared/src/main/scala/lucuma/core/math/Wavelength.scala

```diff
@@ -104,27 +106,28 @@ object Wavelength {
    * Try to build a Wavelength with a value in nm in the range (0 .. 2147]
    * @group constructor
    */
-  def fromMicrometers(µm: Int): Option[Wavelength] =
-    refineV[Positive](µm).toOption.flatMap(µm =>
-      Option.when(µm.value <= MaxMicrometer)(Wavelength(µm.withUnit[Micrometer]))
-    )
+  def fromMicrometers(µm: Int): Option[Wavelength] = for {
+    q <- refineQV[Positive](µm.withUnit[Micrometer].tToUnit[Picometer]).toOption
```

Comment:
What is `refineQV`?

Reply:
A way to refine the value of a `Quantity`.

https://github.com/gemini-hlsw/lucuma-core/blob/3ac15623099b1b854029edbfd3a2cc1780e9f99c/modules/core/shared/src/main/scala/lucuma/core/math/units.scala#L155-L161